### PR TITLE
Adding caching for adapter KVs

### PR DESCRIPTION
**Pull request by gkroiz**
This PR adds caching for `ak` and `av` (adapter keys and adapter values, respectively).

Based on #51 & #58

### CODE REVIEW

**Review by lantiga — APPROVED** [review]
Looks great!

File: lit_parrot/adapter.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def __init__(self, config: Config, block_idx: int) -> None:
             self.adapter_wte = nn.Embedding(config.adapter_prompt_length, config.n_embd)
             # gate for adaption
             self.gating_factor = torch.nn.Parameter(torch.zeros(1))
+            self.adapter_cache_kv = []
```
Comment (carmocca) [inline lit_parrot/adapter.py:46]: Where is this used? The adapter cache kv is saved in Parrot

```suggestion
```

**Review by carmocca — APPROVED** [review]
Nice!

File: lit_parrot/adapter.py (hunk 1 above)
Comment (gkroiz) [reply lit_parrot/adapter.py]: thanks for catching this, I removed this line.

**Pull request merged**